homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -65,190 +63,110 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
-    sensors = [
-        DwdWeatherWarningsSensor(api, name, condition)
-        for condition in config[CONF_MONITORED_CONDITIONS]
-    ]
+    # Build sensor list and activate update only for the first one
+    sensors = []
+    call_update = True
+    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        sensors.append(DwdWeatherWarningsSensor(api, name, sensor_type, call_update))
+        call_update = False
 
     add_entities(sensors, True)
 
 
 class DwdWeatherWarningsSensor(Entity):
     """Representation of a DWD-Weather-Warnings sensor."""
 
-    def __init__(self, api, name, variable):
+    def __init__(self, api, name, sensor_type, call_update):
         """Initialize a DWD-Weather-Warnings sensor."""
         self._api = api
         self._name = name
-        self._var_id = variable
-
-        variable_info = MONITORED_CONDITIONS[variable]
-        self._var_name = variable_info[0]
-        self._var_units = variable_info[1]
-        self._var_icon = variable_info[2]
+        self._sensor_type = sensor_type
+        self._call_update = call_update
 
     @property
     def name(self):
         """Return the name of the sensor."""
-        return f"{self._name} {self._var_name}"
+        return f"{self._name} {MONITORED_CONDITIONS[self._sensor_type][0]}"
 
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        return self._var_icon
+        return MONITORED_CONDITIONS[self._sensor_type][2]
 
     @property
     def unit_of_measurement(self):
         """Return the unit the value is expressed in."""
-        return self._var_units
+        return MONITORED_CONDITIONS[self._sensor_type][1]
 
     @property
     def state(self):
         """Return the state of the device."""
-        try:
-            return round(self._api.data[self._var_id], 2)
-        except TypeError:
-            return self._api.data[self._var_id]
+        if self._sensor_type == "current_warning_level":
+            warning_level = self._api.current_warning_level
+        else:
+            warning_level = self._api.expected_warning_level
+        return warning_level
 
     @property
     def device_state_attributes(self):
         """Return the state attributes of the DWD-Weather-Warnings."""
-        data = {ATTR_ATTRIBUTION: ATTRIBUTION, "region_name": self._api.region_name}
-
-        if self._api.region_id is not None:
-            data["region_id"] = self._api.region_id
+        data = {ATTR_ATTRIBUTION: ATTRIBUTION}
 
-        if self._api.region_state is not None:
-            data["region_state"] = self._api.region_state
-
-        if self._api.data["time"] is not None:
-            data["last_update"] = dt_util.as_local(
-                dt_util.utc_from_timestamp(self._api.data["time"] / 1000)
-            )
+        data["region_name"] = self._api.warncell_name
+        data["region_id"] = self._api.warncell_id
+        data["region_state"] = "N/A"  # Not available via new API
```

Comment:
Is this a breaking change? It needs to be noted if so.

Reply:
Additionally, don't store a string value here – set it to `None`.

Reply:
I think nobody uses the value of region_state because it doesn't change if you don't select another region_name in your configuration.
My first thougth was to completely remove the region_state attribute and mark it as breaking change. My secound thougth was to keep it but set it to "N/A" and don't mark it as breaking change.
What do you think is the right approach?

Reply:
If it isn't useful, remove it and label it a breaking change.

Reply:
Removed and labeled as breaking change.